_source/_posts/2021-12-06-spring-data-elasticsearch.md

```diff
@@ -0,0 +1,369 @@
+
+
+```shell
+source .okta.env
+echo \"
+spring:
+  security:
+    oauth2:
+      client:
+        provider:
+          oidc:
+            issuer-uri: $SPRING_SECURITY_OAUTH2_CLIENT_PROVIDER_OIDC_ISSUER_URI
+        registration:
+          oidc:
+            client-id: $SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_ID
+            client-secret: $SPRING_SECURITY_OAUTH2_CLIENT_REGISTRATION_OIDC_CLIENT_SECRET\" >> docker-compose/central-server-config/application.yml
+```
+
+
+For the experiment in this tutorial, configure the Kibana interface for Elasticsearch, which will allow visualizations of the Elasticsearch data.
+Edit `docker-compose/docker-compose.yml` and add the Kibana service like this:
+
+```yml
+services:
+  ...  
+  blog-kibana:
+     image: docker.elastic.co/kibana/kibana:7.15.2
+     ports:
+       - 5601:5601
+     environment:
+       ELASTICSEARCH_URL: http://blog-elasticsearch:9200
+       ELASTICSEARCH_HOSTS: '["http://blog-elasticsearch:9200"]'
+```
+**NOTE**: The Kibana version must match the version of Elasticsearch that's used by the `blog-elasticsearch` image.
+
+Also add the following additional configuration to the `blog-elasticsearch` service, to expose the ports to the local host for sending APIs requests:
+
+```yml
+services:
+  blog-elasticsearch:
+    ...
+    ports:
+      - 9200:9200
+```
+Remove the Keycloak service at the bottom, as Okta will be used as Identity Provider.
+
+Create each applications' container image:
+
+```shell
+cd blog
+./mvnw -DskipTests -ntp -Pprod verify jib:dockerBuild
+```
+
+```shell
+cd gateway
+./mvnw -DskipTests -ntp -Pprod verify jib:dockerBuild
+```
+
+Run all your images with Docker Compose:
+
+```shell
+cd ../docker-compose
+docker compose up
+```
+ Access the JHipster Registry at `http://localhost:8761` and sign in with your Okta credentials. When you see all services up and green, go to `http://localhost:8080` and sign in with your Okta account.
+
+ {% img blog/spring-data-elasticsearch/okta-signin.png alt:"Okta Sign In Form" width:"400" %}{: .center-image }
+
+
+ **IMPORTANT NOTE**: There is a potential issue observed for the blog service, as [Kubernetes probes](https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.endpoints.kubernetes-probes) are enabled by default, the blog application might change its state to `OUT_OF_SERVICE` during start up and Eureka discards the following heartbeats with state `UP`. In that case, disable Kubernetes probes in the `docker-compose.yml` with the following environment variable:
```

Comment:
```suggestion
 **IMPORTANT NOTE**: There is a potential issue observed for the blog service. As [Kubernetes probes](https://docs.spring.io/spring-boot/docs/current/reference/html/actuator.html#actuator.endpoints.kubernetes-probes) are enabled by default, the blog application might change its state to `OUT_OF_SERVICE` during start up and Eureka might discard the following heartbeats with state `UP`. In this case, disable Kubernetes probes in the `docker-compose.yml` with the following environment variable:
```
simplify by making two sentences. Should "state `UP`"  be "start `UP`" in this sentence?

Reply:
Hi Havi, no, the value of the "state" is "UP"